src/shared/store/authStore.ts

```diff
@@ -0,0 +1,23 @@
+import { create } from 'zustand'
+import { persist } from 'zustand/middleware'
+import type { User } from '@shared/types'
+
+interface AuthState {
+  user: User | null
+  setUser: (user: User) => void
+}
+
+const useAuthStore = create<AuthState>()(
+  persist(
+    (set) => ({
+      user: null,
+      setUser: (user) => set({ user }),
+    }),
+    {
+      name: 'auth-user',
+      getStorage: () => localStorage,
```

Comment:
지원이 안되는 메서드를 사용하셨네요.

`storage: createJSONStorage(() => localStorage)` 이렇게 바꿔주셔야합니다

Reply:
어제 급하게 만든다고 블로그 글보고 했더니 옛날 글이었나보네요..찾아보니 기본적으로 zustand persist는 localStorage를 사용하기 때문에, 따로 명시하지 않아도 된다고 하네요!

요렇게 바꾸겠습니다.

```suggestion
    {
      name: 'auth-user',
    },
```